fe/fe-core/src/main/java/com/starrocks/connector/PartitionUtil.java

```diff
@@ -793,4 +794,37 @@ public static String getPartitionName(String basePath, String partitionPath) {
     public static String getPathWithSlash(String path) {
         return path.endsWith("/") ? path : path + "/";
     }
+
+    /**
+     * <p>
+     *     In the Iceberg Partition Evolution scenario, 'org.apache.iceberg.PartitionField#name' only represents the
+     *     name of a partition in the Iceberg table's Partition Spec. This name is used when trying to obtain the
+     *     names of Partition Spec partitions. e.g.
+     * </p>
+     * <p>
+     *     {
+     *   "source-id": 4,
+     *   "field-id": 1000,
+     *   "name": "ts_day",
+     *   "transform": "day"
+     *   }
+     * </p>
+     * <p>
+     *     column id is '4', column name is 'ts', but 'PartitionField#name' is 'ts_day', 'PartitionField#fieldId'
+     *     is '1000', 'PartitionField#name' default is 'columnName_transformName', and we can customize this name.
+     *     So even for an Identity Transform, this name doesn't necessarily have to match the schema column name,
+     *     because we can customize this name. But in general, nobody customize an Identity Transform Partition name.
+     * </p>
+     * <p>
+     *     To obtain the table columns for Iceberg tables, we use 'org.apache.iceberg.Schema#findColumnName'.
+     * </p>
+     *<br>
+     * refs:<br>
+     * - https://iceberg.apache.org/spec/#partition-evolution<br>
+     * - https://iceberg.apache.org/spec/#partition-specs<br>
+     * - https://iceberg.apache.org/spec/#partition-transforms
+     */
+    public static String getPartitionSourceName(Schema schema, PartitionField partition) {
```

Comment:
You could put this at IcebergPartitionUtils

Reply:
done